src/Microsoft.Framework.FileSystemGlobbing/Infrastructure/MatcherContext.cs

```diff
@@ -0,0 +1,277 @@
+
+        public FrameData Frame;
+        public Stack<FrameData> FrameStack = new Stack<FrameData>();
+
+        public PatternMatchingResult Execute()
+        {
+            Files = new List<string>();
+
+            Frame.Stage = Stage.Complete;
+            PushFrame(Stage.Predicting, DirectoryInfo);
+
+            while (Frame.Stage != Stage.Complete)
+            {
+                DoPredicting();
+                DoEnumerating();
+                DoRecursion();
+            }
+
+            return new PatternMatchingResult(Files);
+        }
+
+        private void DoPredicting()
+        {
+            if (Frame.Stage != Stage.Predicting)
+            {
+                return;
+            }
+            //foreach (var patternContext in IncludePatternContexts)
+            //{
+            //    patternContext.PredictInclude();
+            //}
+            //foreach (var patternContext in ExcludePatternContexts)
+            //{
+            //    //patternContext.PredictExclude();
+            //}
+            Frame.Stage = Stage.Enumerating;
+        }
+
+        private void DoEnumerating()
+        {
+            if (Frame.Stage != Stage.Enumerating)
+            {
+                return;
+            }
+            foreach (var fileSystemInfo in Frame.DirectoryInfo.EnumerateFileSystemInfos(
+                "*",
+                SearchOption.TopDirectoryOnly))
+            {
+                var directoryInfo = fileSystemInfo as DirectoryInfoBase;
+                if (directoryInfo != null)
+                {
+                    var include = false;
+                    foreach (var pattern in IncludePatternContexts)
+                    {
+                        if (pattern.Test(directoryInfo))
+                        {
+                            include = true;
+                            break;
+                        }
+                    }
+                    if (include)
+                    {
+                        foreach (var pattern in ExcludePatternContexts)
+                        {
+                            if (pattern.Test(directoryInfo))
+                            {
+                                include = false;
+                                break;
+                            }
+                        }
+                    }
+                    if (include)
+                    {
+                        if (Frame.ActualDirectories == null)
+                        {
+                            Frame.ActualDirectories = new List<DirectoryInfoBase>();
+                        }
+                        Frame.ActualDirectories.Add(directoryInfo);
+                    }
+                    continue;
+                }
+                var fileInfo = fileSystemInfo as FileInfoBase;
+                if (fileInfo != null)
+                {
+                    var include = false;
+                    foreach (var pattern in IncludePatternContexts)
```

Comment:
Please check this PR for the update https://github.com/aspnet/FileSystem/pull/52